File: go/tree/654. 最大二叉树.go
package tree

import "math"

func constructMaximumBinaryTree(nums []int) *TreeNode {
	if len(nums) == 0 {
		return nil
	}
	i := getMaxIndex(nums)
	node := &TreeNode{Val: nums[i]}
	node.Left = constructMaximumBinaryTree(nums[:i])
	node.Right = constructMaximumBinaryTree(nums[i+1:])
	return node
}

func getMaxIndex(A []int) int {
	index, max := 0, math.MinInt32
	for i, v := range A {
		if v > max {
			max = v
			index = i
		}
	}
	return index
}

File: python/树/654. 最大二叉树.py
from typing import List

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def constructMaximumBinaryTree(self, nums: List[int]) -> TreeNode:
        if not nums:
            return
        i = self.getIndex(nums)
        root = TreeNode(nums[i])
        root.left = self.constructMaximumBinaryTree(nums[:i])
        root.right = self.constructMaximumBinaryTree(nums[i+1:])
        return root
    def getIndex(self,nums):
        index,max_num = 0,float('-inf')
        for i in range(len(nums)):
            if nums[i] > max_num:
                index = i
                max_num = nums[i]
        return index
